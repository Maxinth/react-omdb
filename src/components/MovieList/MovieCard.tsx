import { Card } from "./movieList.styled";
import { ListItem } from ".";
import React from "react";

const MovieCard = (props: ListItem) => {
  const { Title } = props;
  return (
    <Card
    //  style={{ backgroundImage: `url(${Poster})` }}
    >
      {Title}
    </Card>
  );
};

export default MovieCard;
