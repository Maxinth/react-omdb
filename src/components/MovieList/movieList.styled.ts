import styled from "styled-components";

const Container = styled.section`
  align-self: flex-start;
  margin-top: 33px;
  margin-bottom: 26px;
  /* background-color: #e5e5e5; */
  /* display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column; */
  padding-left: 28px;
  /* padding-right: 27px; */
  @media (min-width: 834px) {
    /* padding-right: 77px; */
    padding-left: 77px;
  }

  @media (min-width: 1440px) {
    /* padding-right: 57px; */
  }
`;
const Title = styled.p`
  text-align: left;
  width: 210px;
  height: 23px;
  left: 28px;
  top: 472px;
  margin-bottom: 26px;
  font-family: "DM Sans", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 23px;
  transition: all 0.2s;
  color: #000000;

  @media (min-width: 834px) {
    font-family: "DM Sans", sans-serif;
    font-size: 24px;
    font-style: normal;
    font-weight: 400;
    line-height: 31px;
    letter-spacing: 0em;
    text-align: left;
    margin-bottom: 18px;
    height: 31px;
    width: 100%;
  }
`;

const InnerBox = styled.div``;
const MoviesBox = styled.section`
  overflow-x: scroll;

  //hide the scrollbar
  ::-webkit-scrollbar {
    width: 12px;
  }
  ::-webkit-scrollbar-track {
    display: none;
  }
  //hide the scrollbar
  ${InnerBox} {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
`;
const Card = styled.div`
  min-height: 200px;
  min-width: 200px;
  max-height: 200px;
  max-width: 200px;
  border-radius: 12px;
  padding: 10px;
  background-color: #000000;
  color: #fff;
  display: flex;
  font-weight: bolder;
  justify-content: center;
  align-items: center;
  margin-right: 13px;
  font-size: 18px;
  transition: all 0.2s;
  text-align: center;
  font-family: "DM Sans", sans-serif;
  @media (min-width: 834px) {
    min-width: 300px;
    min-height: 300px;
    max-width: 300px;
    max-height: 300px;
    font-size: 24px;
  }
`;
export { Container, Title, Card, MoviesBox, InnerBox };
