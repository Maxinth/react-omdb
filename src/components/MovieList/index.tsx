import MovieCard from "./MovieCard";
import { Container, Title, MoviesBox, InnerBox } from "./movieList.styled";

export interface ListItem {
  Poster: string;
  Title: string;
  Type: string;
  Year: string;
  imdbID: string;
}

interface Props {
  list: ListItem[];
  category: string;
}
const MovieList = ({ list, category }: Props) => {
  // const
  return (
    <Container>
      <Title>{category}</Title>
      <MoviesBox>
        <InnerBox>
          {list?.map((item: ListItem, index: number) => (
            <MovieCard key={index} {...item} />
          ))}
        </InnerBox>
      </MoviesBox>
    </Container>
  );
};

export default MovieList;
