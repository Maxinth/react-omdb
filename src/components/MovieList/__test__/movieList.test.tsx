import MovieList from "..";
import renderer from "react-test-renderer";
import React from "react";

const mockProps = {
  list: [
    {
      Poster: "me",
      Title: "snapshots",
      Type: "first-type",
      Year: "2022",
      imdbID: "---",
    },
  ],
  category: "FC",
};

it("renders a snapshot of <MovieList with mock props", () => {
  const elem = renderer.create(<MovieList {...mockProps} />).toJSON();
  expect(elem).toMatchSnapshot();
});
