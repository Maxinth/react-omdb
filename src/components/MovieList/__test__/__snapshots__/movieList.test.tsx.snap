// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <MovieList with mock props 1`] = `
<section
  className="sc-bdvvtL kxrhlG"
>
  <p
    className="sc-gsDKAQ fsJgEz"
  >
    FC
  </p>
  <section
    className="sc-hKwDye ixHXeB"
  >
    <div
      className="sc-dkPtRN"
    >
      <div
        className="sc-eCImPb iTldMi"
      >
        snapshots
      </div>
    </div>
  </section>
</section>
`;
